include/accessor.h

```diff
@@ -407,6 +407,25 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
 
 	template <access_mode M = Mode>
 	inline std::enable_if_t<detail::access::mode_traits::is_producer(M), DataT&> operator[](const id<Dims>& index) const {
+		
+	// analogue to the device buffer. 
+	// without sycl::memory ...
```

Comment:
```suggestion
```
I think this is clear from context.

However come to think of it, we might have to use atomics on the host as well. While there is no issue between multiple chunks running concurrently (they receive different `accessor` instances), there's nothing stopping the user from doing manual multi-threading inside a host task, therefore IMO `operator[]` should be thread safe. Thoughts @PeterTh, @fknorr ?

Reply:
I agree on atomics. As a user I would deem read-accessors thread safe. Unfortunately we don't have `std::atomic_ref` in C++17, but something along the lines of `std::vector<std::unique_ptr<std::array<std::atomic<size_t>, 3>>>` should do.